GenderPayGap.WebUI/Controllers/Admin/AdminDatabaseIntegrityChecksController.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using GenderPayGap.Core;
+using GenderPayGap.Core.Classes;
+using GenderPayGap.Core.Interfaces;
+using GenderPayGap.Database;
+using GenderPayGap.WebUI.Models.Admin;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace GenderPayGap.WebUI.Controllers.Admin
+{
+    [Authorize(Roles = "GPGadmin")]
+    [Route("admin")]
+    public class AdminDatabaseIntegrityChecksController : Controller
+    {
+
+        private readonly IDataRepository dataRepository;
+
+        public AdminDatabaseIntegrityChecksController(
+            IDataRepository dataRepository)
+        {
+            this.dataRepository = dataRepository;
+        }
+
+        [HttpGet("database-integrity-checks")]
+        public async Task<IActionResult> DatabaseIntegrityChecks()
+        {
+            return View("DatabaseIntegrityChecks");
+        }
+
+        [HttpGet("organisations-with-multiple-active-addresses-ajax")]
+        public async Task<IActionResult> OrganisationsWithMultipleActiveAddressesAjax()
+        {
+            var model = new DatabaseIntegrityChecksViewModel();
+            List<Organisation> organisations = dataRepository.GetAll<Organisation>()
+                .Include(o => o.OrganisationAddresses)
+                .ToList();
+
+            foreach (Organisation organisation in organisations)
+            {
+                IEnumerable<OrganisationAddress> activeAddresses = organisation.OrganisationAddresses
+                    .Where(oa => oa.Status == AddressStatuses.Active);
+                if (activeAddresses.Count() > 2)
```

Comment:
(not important)
If you want, you can simplify
`enumerable.Where(x => x.Foo).Count()`
to
`enumerable.Count(x => x.Foo)`

Reply:
Thanks